builder/dockerfile/shell_parser.go

```diff
@@ -169,25 +170,23 @@ func (sw *shellWord) processSingleQuote() (string, error) {
 	//   all the characters (except single quotes, making it impossible to put
 	//   single-quotes in a single-quoted string).
 
-	var result string
+	var result bytes.Buffer
 
 	sw.scanner.Next()
 
+loop:
```

Comment:
And if the only reason we're doing it is to switch between an IF and a CASE I don't see the benefit.